RockWeb/Webhooks/WebhookToWorkflow.ashx

```diff
@@ -0,0 +1,394 @@
+    /// <returns>true if the DefinedValue matches this request, otherwise false.</returns>
+    protected virtual bool IsHookValidForRequest( DefinedValueCache hook )
+    {
+        string hookUrl = hook.GetAttributeValue( "Url" );
+        string hookMethod = hook.GetAttributeValue( "Method" );
+
+        //
+        // Check for match on method type, if not continue to the next item.
+        //
+        if ( !string.IsNullOrEmpty( hookMethod ) && !HttpContext.Request.HttpMethod.ToString().Equals( hookMethod, StringComparison.InvariantCultureIgnoreCase ) )
+        {
+            return false;
+        }
+
+        //
+        // Check for match on the URL.
+        //
+        if ( string.IsNullOrEmpty( hookUrl ) )
+        {
+            return true;
+        }
+        else if ( hookUrl.StartsWith( "^" ) && hookUrl.EndsWith( "$" ) )
+        {
+            return Regex.IsMatch( Url, hookUrl, RegexOptions.IgnoreCase );
+        }
+        else
+        {
+            return Url.Equals( hookUrl, StringComparison.InvariantCultureIgnoreCase );
+        }
+    }
+
+    /// <summary>
+    /// Populates any defined Workflow attributes specified to this webhook type.
+    /// Subclasses may call the base method to have the Request attribute set.
+    /// </summary>
+    /// <param name="workflow">The workflow whose attributes need to be set.</param>
+    /// <param name="hook">The DefinedValue of the currently executing webhook.</param>
+    protected virtual void PopulateWorkflowAttributes( Workflow workflow, DefinedValueCache hook )
+    {
+        workflow.SetAttributeValue( "Request", JsonConvert.SerializeObject( RequestToDictionary( hook ) ) );
+
+        string workflowAttributes = hook.GetAttributeValue( "WorkflowAttributes" );
+        string nameTemplate = hook.GetAttributeValue( "WorkflowNameTemplate" );
+
+        string[] attributes = workflowAttributes.Split( '|' );
+
+        foreach ( string attribute in attributes )
+        {
+            if ( attribute.Contains( '^' ) )
+            {
+                string[] settings = attribute.Split( '^' );
+                workflow.SetAttributeValue( settings[0], settings[1] );
+            }
+        }
+
+        // set workflow name
+        if ( !string.IsNullOrEmpty( nameTemplate ) )
+        {
+            workflow.Name = nameTemplate;
+        }
+    }
+
+    /// <summary>
+    /// Convert the request into a generic JSON object that can provide information
+    /// to the workflow. If a subclass does needs to customize this data they can
+    /// call the base method and then modify the content before returning it.
+    /// </summary>
+    /// <param name="hook">The DefinedValue of the currently executing webhook.</param>
+    /// <returns></returns>
+    protected virtual Dictionary<string, object> RequestToDictionary( DefinedValueCache hook )
+    {
+        var dictionary = new Dictionary<string, object>();
+
```

Comment:
Use this instead so that dictionary includes a lot of our common merge fields.
```
var dictionary = Rock.Lava.LavaHelper.GetCommonMergeFields( null );
```